xbmc/addons/AddonManager.cpp

```diff
@@ -577,27 +551,43 @@ bool CAddonMgr::GetInstalledBinaryAddons(BINARY_ADDON_LIST& binaryAddonList)
 
   for (auto builder : builders)
   {
-    cp_status_t status;
-    cp_plugin_info_t* cp_addon = m_cpluff->get_plugin_info(m_cp_context, builder.GetId().c_str(), &status);
-    if (status == CP_OK && cp_addon)
+    BINARY_ADDON_LIST_ENTRY binaryAddon;
+    if (GetInstalledBinaryAddon(builder.GetId(), binaryAddon))
+      binaryAddonList.push_back(std::move(binaryAddon));
+  }
+
+  return !binaryAddonList.empty();
+}
+
+bool CAddonMgr::GetInstalledBinaryAddon(const std::string& addonId, BINARY_ADDON_LIST_ENTRY& binaryAddon)
+{
+  bool ret = false;
+  cp_status_t status;
+
+  CSingleLock lock(m_critSection);
+
+  cp_plugin_info_t *cp_addon = m_cpluff->get_plugin_info(m_cp_context, addonId.c_str(), &status);
+  if (status == CP_OK && cp_addon)
+  {
+    cp_extension_t* props = GetFirstExtPoint(cp_addon, ADDON_UNKNOWN);
+    if (props != nullptr)
     {
-      cp_extension_t* props = GetFirstExtPoint(cp_addon, ADDON_UNKNOWN);
-      if (props != nullptr)
+      CAddonBuilder builder;
+//       std::string value = GetPlatformLibraryName(props->plugin->extensions->configuration);
+//       if (!value.empty() &&
+//           props->plugin->plugin_path &&
+//           strcmp(props->plugin->plugin_path, "") != 0 &&
+//           Factory(cp_addon, ADDON_UNKNOWN, builder, true))
```

Comment:
hmm. why are you commenting out instead of actually removing code? If it is a test, please don't push.